### PR TITLE
support CSV import to schema "person"

### LINKED ISSUE #3006: support CSV import to schema "person"
**Describe the bug**
If you'd import a CSV with metadata field person.givenName, it will fail and complain the metadata field doesn't exist.
This is happening due to org.dspace.app.bulkedit.DSpaceCSV#DSpaceCSV(java.io.File, org.dspace.core.Context) which on line 181 calls org.dspace.authority.service.AuthorityValueService#getAuthorityValueType
This will get org.dspace.authority.PersonAuthorityValue#getAuthorityType and treat "person" separately.

This is also a bug in DSpace 6, but is now more problematic because there's a schema with name person.

**To Reproduce**
1. Create a CSV with metadata field person.birthDate or person.givenName
2. Verify the metadata fields exist in the registry
3. Import the CSV, and notice it will fail
4. This can also be verified on a DSpace 6

**Expected behavior**
Authority fields should be treated in a unique way, e.g. on columns containing `[authority]person`

### PR DESCRIPTION
**Pull request by Raf-atmire**
## References
* Fixes https://github.com/DSpace/DSpace/issues/3006

## Description
This PR adds a fix to the DSpaceCSV.java class to make it possible to better distinguish a normal "person" metadata field from an authority field.
Any field that doesn't start with [AUTHORITY_TYPE] is now treated as a regular field without the chance of it being detected as an authorityType, we made sure that this only happens if the String is surrounded by brackets. 

## Instructions for Reviewers

List of changes in this PR:
* Made sure that the AuthorityType evaluating is only done for Strings between brackets
* Added test to ensure that the CSV import with person metadata works now

How to test this PR:
* Make a CSV with a person.birthDate metadata field in it with a value
* Import the CSV and verify that the person.birthDate is added to the imported Item

## Checklist

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

File: dspace-api/src/test/resources/test-config.properties [hunk 1]
```diff
@@ -13,3 +13,4 @@ test.folder = ./target/testing/
 test.bitstream = ./target/testing/dspace/assetstore/ConstitutionofIreland.pdf
 test.exportcsv = ./target/testing/dspace/assetstore/test.csv
 test.importcsv = ./target/testing/dspace/assetstore/testImport.csv
+test.importpersoncsv = ./target/testing/dspace/assetstore/testImportPerson.csv
```
Comment (tdonohue) [inline dspace-api/src/test/resources/test-config.properties:16]: @Raf-atmire : This test file does not appear to be used anywhere?  Am I misunderstanding how/where this configuration is used?  I'd expect it to appear in one of our IT classes.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@Raf-atmire : Gave this a quick look today.  The code looks reasonable, but I'm not sure the tests you've setup are actually running.  I see a test CSV & a test configuration, but where are those used by our ITs?   So, I think this is missing tests right now (unless I'm misunderstanding something).

**Comment by Raf-atmire** [comment]
@tdonohue The test got lost in the latest main merge, I've re-added it. I've also removed the property and the associated csv as they were no longer used. The CSV gets made inline in the test itself like the other tests in MetadataImportIT do. Could you please check this again when you get the time? Thanks!

**Review by benbosman — APPROVED** [review]
I've reviewed this PR, and it looks good.
Also only a very small change with a test

**Review by tdonohue — APPROVED** [review]
👍 Looks good to me now. Test has been added. Thanks @Raf-atmire !

**Comment by tdonohue** [comment]
@Raf-atmire : This looks good now, but now has a merge conflict (after merging #3037 ).  Could you solve the merge conflict?

File: dspace-api/src/test/java/org/dspace/app/bulkedit/MetadataImportIT.java [hunk 2]
```diff
@@ -152,6 +152,22 @@ public void relationshipMetadataImporAlreadyExistingItemTest() throws Exception
...
+        String[] csv = {"id,collection,dc.title,person.birthDate",
+            "+," + collection.getHandle() + ",\"Test Import 2\"," + "2000"};
+        performImportScript(csv);
+        Item importedItem = findItemByName("Test Import 2");
+        assertTrue(
+            StringUtils.equals(
+                itemService.getMetadata(importedItem, "person", "birthDate", null, Item.ANY)
+                           .get(0).getValue(), "2000"));
+        context.turnOffAuthorisationSystem();
+        itemService.delete(context, itemService.find(context, importedItem.getID()));
```
Comment (tdonohue) [inline dspace-api/src/test/java/org/dspace/app/bulkedit/MetadataImportIT.java:167]: Tiny thing I just realized.  You likely can simplify this line to be `itemService.delete(context, importedItem);`  I don't think you should need the `itemService.find()` here...unless I'm missing something.

In any case, this is extremely minor. The code overall is good.

**Comment by Raf-atmire** [comment]
@tdonohue Thank you for the feedback, I've merged the most recent dspace/main into this branch and addressed the minor issue with the itemService.delete in the MetadataImportIT. Could you check again when you get the time please? Thanks!

**Pull request merged**